whereby/sdk
# Add mute participant action

### Description
Add the possibility for **hosts** to mute remote participants. The action `muteParticipants` takes an array of `clientIds`, so you can also implement a "mute all" functionality with this.

**Summary:**



**Related Issue:**



### Testing

1. Check out this branch
2. In the /.env file add a valid host roomKey for the configured STORYBOOK_ROOM in following line:
STORYBOOK_ROOM_HOST_ROOMKEY=[host_roomKey_here]
3. Run yarn build && yarn dev
4. In the Storybook UI (@ http://localhost:6006/) open the story: "Custom UI" > "Room Connection With Host Controls"
5. Join the same room in a different browser (PWA).
6. Clicking on the mute button on the remote participant should mute the participant in the other tab
![image](https://github.com/whereby/sdk/assets/6151521/2a83ca76-9823-4d21-895b-e4ac2caaec0e)
7. Verify that it does in fact mute - you should see this notification in the PWA and the participant should be muted.
![image](https://github.com/whereby/sdk/assets/6151521/23bc3a3d-470a-4e34-8e9e-18224fdc67ba)
8. In the Story controls panel, remove the roomKey and reload the page (Cmd + R).
9. Clicking on the mute button should not trigger a the action. (not connected as a "host" now).
10. Additionally, a warning should be displayed in the console whenever these actions are invoked by a non-"host" user.
![image](https://github.com/whereby/sdk/assets/6151521/a87d099a-1587-4754-84b6-b0db5d995dd0)




### Screenshots/GIFs (if applicable)



### Checklist

        under the terms of the MIT license.

### Additional Information
